### DISCUSSION TITLE
tutorials/intro-dgs/11-resolver-chains

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#157] [course feedback]
# tutorials/intro-dgs/11-resolver-chains

Learn how resolver chains and the source argument works.

https://www.apollographql.com/tutorials/intro-dgs/11-resolver-chains

**Comment by boolivar** [comment]
`ParameterizedTypeReference` can be used in `body()` method for `List` retrieval:

```java
public List<Amenity> amenitiesRequest(String listingId) {
    return client
        .get()
        .uri("/listings/{listing_id}/amenities", listingId)
        .retrieve()
        .body(new ParameterizedTypeReference<List<Amenity>>() {});
}
```
